### PR TITLE
[Feature] CatFrames used as inverse 

### PR DESCRIPTION
**Pull request by BY571**
## Description

Adding the 'as_inverse' option to the CatFrames transform. With this transform we can use the CatFrames transform when adding data to the Replay Buffer. By that, the RB will consist of unfolded sequential samples. 

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [X] I have read the [CONTRIBUTION](https://github.com/pytorch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [X] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: test/test_transforms.py [hunk 1]
```diff
@@ -466,6 +466,29 @@ def test_transform_rb(self, dim, N, padding, rbclass):
...
+        cat_frames = CatFrames(
+            N=N, in_keys=in_keys, dim=dim, padding=padding, as_inverse=True
+        )
+
+        env = TransformedEnv(
+            ContinuousActionVecMockEnv(),
+        )
+        obs_dim = env.specs["output_spec"]["_observation_spec"][
+            "observation_orig"
+        ].shape[0]
```
Comment (vmoens) [inline test/test_transforms.py:485]: why not `env.observation_spec["observation_orig].shape[0]`?

File: torchrl/envs/transforms/transforms.py [hunk 2]
```diff
@@ -1960,6 +1960,7 @@ class CatFrames(ObservationTransform):
             has to be written. Defaults to the value of `in_keys`.
         padding (str, optional): the padding method. One of ``"same"`` or ``"zeros"``.
             Defaults to ``"same"``, ie. the first value is uesd for padding.
+        as_inverse (bool, optional): if ``True``, the transform is applied as an inverse transform.
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:1963]:
```suggestion
        as_inverse (bool, optional): if ``True``, the transform is applied as an inverse transform.
            Defaults to ``False``.
```

**Review by vmoens — COMMENTED** [review]
We should also check what happens when we set `as_inverse=True` and append it to an environment. This should modify the observation in the input tensordict, which is pretty useless unless it is also part of the input keys. 
So either we raise an exception when `as_inverse=True` and the transform has a parent or we make it possible but then we need to define what behaviour is expected and what transformation the observation_spec / input_spec should have. I'm more in favour of the first option (raising an exception if it has a parent).

File: torchrl/envs/transforms/transforms.py [hunk 3]
```diff
@@ -2150,6 +2163,12 @@ def transform_observation_spec(self, observation_spec: TensorSpec) -> TensorSpec
...
 
     def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
+        if self.as_inverse:
+            return tensordict
+        else:
+            return self.unfolding(tensordict)
+
+    def unfolding(self, tensordict: TensorDictBase) -> TensorDictBase:
         # it is assumed that the last dimension of the tensordict is the time dimension
         if not tensordict.ndim or tensordict.names[-1] != "time":
```
Comment (BY571) [inline torchrl/envs/transforms/transforms.py:2173]: Still unsure about this. I know its purpose but not sure if it is really needed. Also if we keep it we probably need a transform that adds the time dimension to a tensordict. Because when using CatFrames as an inverse applied to the replay buffer the data does not have the time dimension. At least for me, it errors out every time.

Comment (vmoens) [reply torchrl/envs/transforms/transforms.py:2173]: What about a warning that says that we assume the last dim to be "time"? The thing we want to avoid it that time is on another dim and the catframe wrongly assumes that the last dim is time.
Another thing we could do: for now raise a warning, in the future change the behaviour and if the "time" dim is defined then we know which dim must be considered time, and if not we assume it's the last.
TLDR now you can just replace this by a warning

**Review by vmoens — COMMENTED** [review]
Almost there, can you add a simple check in `test_transform_as_inverse` where you create the transform and append it to an env, and check that the error is raised appropriately?

**Review by vmoens — APPROVED** [review]
LGTM (final)
